### ISSUE TITLE
KeyError: 'action' in _simplify_filters, if server-side filter has no "Do this:" action

### ISSUE THREAD

**Issue by dwhughes** [mesozoic/gmail-yaml-filters#5]
Encountered in versions: 0.6.1, 0.7.1.

Steps to reproduce:
1. Create a rule in filters.yaml: `- { label: Bug demo, from: anywhere@example.com }`
2. `gmail-yaml-filters --sync filters.yaml`
3. Delete the "Bug demo" label via the GMail interface. (The label will be removed from the rule, so you now have a rule that matches anywhere@example.com but has no action.)
4. `gmail-yaml-filters --sync filters.yaml`

Resulting stacktrace:

```
Traceback (most recent call last):
  File "/usr/local/google/home/marain/hack/gmail-filters/venv/bin/gmail-yaml-filters", line 11, in <module>
    sys.exit(main())
  File "/usr/local/google/home/marain/hack/gmail-filters/venv/local/lib/python2.7/site-packages/gmail_yaml_filters/main.py", line 606, in main
    upload_ruleset(ruleset, service=gmail, dry_run=args.dry_run)
  File "/usr/local/google/home/marain/hack/gmail-filters/venv/local/lib/python2.7/site-packages/gmail_yaml_filters/upload.py", line 179, in upload_ruleset
    known_filters = GmailFilters(service)
  File "/usr/local/google/home/marain/hack/gmail-filters/venv/local/lib/python2.7/site-packages/gmail_yaml_filters/upload.py", line 146, in __init__
    self.reload()
  File "/usr/local/google/home/marain/hack/gmail-filters/venv/local/lib/python2.7/site-packages/gmail_yaml_filters/upload.py", line 150, in reload
    self.matchable_filters = [_simplify_filter(existing) for existing in self.filters]
  File "/usr/local/google/home/marain/hack/gmail-filters/venv/local/lib/python2.7/site-packages/gmail_yaml_filters/upload.py", line 139, in _simplify_filter
    'action': {key: set(values) for key, values in filter_dict['action'].items()},
KeyError: 'action'
```

Workaround: manually delete the rule, and re-sync.

**Comment by mesozoic** [comment]
Thanks for pointing this out! I'm sorry for the delay in responding, but it turns out I needed to tune my GitHub notification settings. It does look like an edge case I didn't anticipate and easily fixed with `get` vs `[]`; I'll work on a test and a patch soon but I'm open to pull requests if anyone else can get to it sooner. :)

**Comment by mesozoic** [comment]
Thanks for reporting this! I've released v0.8.1 which should fix this bug.
